pkg/occlient/occlient_test.go

```diff
@@ -99,6 +100,121 @@ func fakePodStatus(status corev1.PodPhase, podName string) *corev1.Pod {
 	}
 }
 
+func TestAddPVCToDeploymentConfig(t *testing.T) {
+	type args struct {
+		dc   *appsv1.DeploymentConfig
+		pvc  string
+		path string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Test case : valid dc",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+						Template: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name: "test",
+										VolumeMounts: []corev1.VolumeMount{
+											{
+												MountPath: "/tmp",
+												Name:      "test",
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				pvc:  "testglocks volume",
+				path: "/mnt",
+			},
+			wantErr: false,
+		},
+		{
+			name: "Test case: dc without Containers defined",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+						Template: &corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{},
+						},
+					},
+				},
+				pvc:  "my xx voulme",
+				path: "/mnt",
+			},
+			wantErr: true,
+		},
+		{
+			name: "Test case: dc without Template defined",
+			args: args{
+				dc: &appsv1.DeploymentConfig{
+					Spec: appsv1.DeploymentConfigSpec{
+						Selector: map[string]string{
+							"deploymentconfig": "nodejs-app",
+						},
+					},
+				},
+				pvc:  "my xx voulme",
+				path: "/mnt",
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient, fakeClientset := FakeNew()
+
+			fakeClientset.AppsClientset.PrependReactor("update", "deploymentconfigs", func(action ktesting.Action) (bool, runtime.Object, error) {
+				return true, nil, nil
```

Comment:
Also we should check if the update reactor was called with the correct dc name passed to the function like https://github.com/redhat-developer/odo/blob/e8e223b6edd6f046035ec1ad249448303fbf4b67/pkg/occlient/occlient_test.go#L1323-L1326 because we might end up calling update on the wrong dc.